packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -385,3 +388,51 @@ export function mapOpenSeaContractV2ToV1(
     },
   };
 }
+
+/**
+ * Retrieves token prices for a set of contract addresses in a specific currency and chainId.
+ *
+ * @param args - The arguments to function.
+ * @param args.tokenPricesService - An object in charge of retrieving token prices.
+ * @param args.nativeCurrency - The native currency to request price in.
+ * @param args.tokenAddresses - The list of contract addresses.
+ * @param args.chainId - The chainId of the tokens.
+ * @returns The prices for the requested tokens.
+ */
+export async function fetchTokenContractExchangeRates({
+  tokenPricesService,
+  nativeCurrency,
+  tokenAddresses,
+  chainId,
+}: {
+  tokenPricesService: AbstractTokenPricesService;
+  nativeCurrency: string;
+  tokenAddresses: Hex[];
+  chainId: Hex;
+}): Promise<ContractExchangeRates> {
+  const isChainIdSupported =
+    tokenPricesService.validateChainIdSupported(chainId);
+  const isCurrencySupported =
+    tokenPricesService.validateCurrencySupported(nativeCurrency);
+
+  if (!isChainIdSupported || !isCurrencySupported) {
+    return {};
+  }
+
+  const tokenPricesByTokenContractAddress =
+    await tokenPricesService.fetchTokenPrices({
```

Comment:
I just thought of something. We recently added code to TokenRatesController which breaks up tokens into 1000-length batches and makes requests that way. You can see this happening here: https://github.com/MetaMask/core/blob/76787ceeab5083780f50a2f9e68617f08cc600c4/packages/assets-controllers/src/TokenRatesController.ts#L515. However, this function would not do that. Should we copy that code here as well? As a plus we can then refactor TokenRatesController later on to use this function and everything should work the same way. Sorry for not recommending this earlier.

Reply:
Hey @mcmire 👋  thanks a lot for reviewing again;

I made the update to make the fct support fetching in batches 🙏  
I guess initially this function was intended to be used in the import tokens flow, where you would expect users to select just a couple of tokens to import, you wouldn't expect it to go up to 100. 

But refactoring this in the future, and have tokenRatesController use this new utility fct is a good reason to have it more generic 🙌